core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/udf/r/RUDFSourceOpDesc.java

```diff
@@ -0,0 +1,117 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import static java.util.Collections.singletonList;
+import static scala.jdk.javaapi.CollectionConverters.asScala;
+
+
+public class RUDFSourceOpDesc extends SourceOperatorDescriptor {
+    @JsonProperty(
+        required = true,
+        defaultValue =
+            "# Input: None\n" +
+            "# Output/Return: Table-like object that can be split into tuples\n" +
+            "function() {\n\n" +
+            "}"
+    )
+    @JsonSchemaTitle("R Source UDF Script")
+    @JsonPropertyDescription("Input your code here")
+    public String code;
+
+    @JsonProperty(required = true)
+    @JsonSchemaTitle("Worker count")
+    @JsonPropertyDescription("Specify how many parallel workers to lunch")
+    public Integer workers = 1;
+
+    @JsonProperty()
+    @JsonSchemaTitle("Columns")
+    @JsonPropertyDescription("The columns of the source")
+    public List<Attribute> columns;
+
+    @Override
+    public PhysicalOp getPhysicalOp(WorkflowIdentity workflowId, ExecutionIdentity executionId) {
+        OpExecInitInfo exec = OpExecInitInfo.apply(code, "r");
+        Preconditions.checkArgument(workers >= 1, "Need at least 1 worker.");
+        SchemaPropagationFunc func = SchemaPropagationFunc.apply((Function<Map<PortIdentity, Schema>, Map<PortIdentity, Schema>> & Serializable) inputSchemas -> {
+            // Initialize a Java HashMap
+            java.util.Map<PortIdentity, Schema> javaMap = new java.util.HashMap<>();
+
+            javaMap.put(operatorInfo().outputPorts().head().id(), sourceSchema());
+
+            // Convert the Java Map to a Scala immutable Map
+            return AmberUtils.toImmutableMap(javaMap);
+        });
+        PhysicalOp physicalOp = PhysicalOp.sourcePhysicalOp(
+                        workflowId,
+                        executionId,
+                        operatorIdentifier(),
+                        exec
+                )
+                .withInputPorts(operatorInfo().inputPorts())
+                .withOutputPorts(operatorInfo().outputPorts())
+                .withIsOneToManyOp(true)
+                .withPropagateSchema(func)
+                .withLocationPreference(Option.empty());
+
+
+        if (workers > 1) {
+            return physicalOp
+                    .withParallelizable(true)
+                    .withSuggestedWorkerNum(workers);
+        } else {
+            return physicalOp.withParallelizable(false);
+        }
+
+    }
+
+    @Override
+    public OperatorInfo operatorInfo() {
+        return new OperatorInfo(
+                "1-out R UDF",
+                "User-defined function operator in R script",
+                OperatorGroupConstants.UDF_GROUP(),
+                asScala(new ArrayList<InputPort>()).toList(),
+                asScala(singletonList(new OutputPort(new PortIdentity(0, false), "", false))).toList(),
+                false,
+                false,
+                true,
```

Comment:
Please add the key for those parameters. I think probably all of them should be false